thrift/matching.thrift

```diff
@@ -149,6 +149,7 @@ service MatchingService {
       2: shared.InternalServiceError internalServiceError,
       3: shared.LimitExceededError limitExceededError,
       4: shared.ServiceBusyError serviceBusyError,
+      5: shared.TaskListNotOwnedByHostError taskListNotOwnedByHostError,
```

Comment:
what happens to the old clients (workers that is deployed before this change) when they receive this error? 
For some reason worker code is very strict about responses and can cause crashes for unrecognized enums. Not sure what the behavior is for unrecognized exceptions. It's worth to try

Reply:
The error will not be returned from the Poll requests. An empty response will be returned instead (updated here and in the PR in the server repo). 

In general frontend will convert this error to a generic system error as frontend does not know about it. If this error after retrys etc is still returned, it should be considered a generic system error, so that is what we want.